src/julia.h

```diff
@@ -232,6 +232,21 @@ typedef struct _jl_line_info_node_t {
     intptr_t inlined_at;
 } jl_line_info_node_t;
 
+typedef union {
```

Comment:
Some compilers may be unhappy with the anonymous decl. IIRC, that is not valid in some standard.
```suggestion
typedef union _u_jl_purity_overrides_t {
```

Reply:
might instead be worth only declaring the struct portion (to avoid the anonymous unions), and do the cast `*(char*)&struct` when required on the load/store?